src/searcher.rs

```diff
@@ -194,13 +202,17 @@ impl Searcher {
 /// It consists in an id identifying its segment, and its segment-local DocId.
 /// The id used for the segment is actually an ordinal in the list of segment
 /// hold by a Searcher.
-#[pyclass]
-#[derive(Clone, Debug)]
+#[pyclass(module = "tantivy")]
+#[derive(Clone, Debug, PartialEq)]
 pub(crate) struct DocAddress {
     pub(crate) segment_ord: tv::SegmentOrdinal,
     pub(crate) doc: tv::DocId,
 }
 
+impl_py_copy!(DocAddress);
```

Comment:
I think rather than support copying, `DocAddress` should become a `#[pyclass(frozen)]` because it is immutable and hence references can be copied freely without any shared mutability issues, i.e. there should never be a need for deep copying.

Reply:
I didn't realise that was possible, this is much better yes.